packages/kmod-6.1-nvidia-r570/grid-license-check.service

```diff
@@ -0,0 +1,26 @@
+[Unit]
+Description=GRID License Check
+RefuseManualStart=true
+RefuseManualStop=true
+DefaultDependencies=no
+Before=kubelet.service
+After=nvidia-gridd.service
+Requires=nvidia-gridd.service
+
+[Service]
+Type=oneshot
+ExecCondition=/usr/bin/ghostdog match-nvidia-driver grid
+# Otherwise, attempt to load the module.
+ExecStart=/usr/bin/nvidia-smi -q
+# Ensure that the stderr file exists. Otherwise, grep fails on an empty file.
+ExecStart=-/usr/bin/touch /tmp/.nvidia-gridd-license
+# Succeed unless there was a fatal error.
+ExecStart=/usr/bin/grep -Fqvzw Unlicensed /tmp/.nvidia-gridd-license
+RemainAfterExit=true
+StandardOutput=append:/tmp/.nvidia-gridd-license
+Restart=on-failure
+RestartSec=1
+StartLimitBurst=120
+
+[Install]
+RequiredBy=kubelet.service
```

Comment:
Below it's required by `nvidia-k8s-device-plugin.service` which seems more correct.

Reply:
Sorry, this was originally `nvidia-k8s-device-plugin.service` and I missed the refactor. In my mind we have several options for what experience we want:
* Don't prevent anything but log it. Sort of the behavior we have today just without logs (not ideal)
* Prevent kubelet from starting if the license isn't valid. This means the node never becomes ready and cannot accept any orchestrated work. (This is what I was intending to enforce in the PR)
* Prevent the device plugin from running, the node will become ready but not advertise GPU resources. It could take some work but not GPU work. (This might be confusing to users so I leaned away from this but was my original approach)
* Attempt to block boot entirely so the node doesn't even reach multi-user. (Seems harsh but would be fine too).